src/components/HFUI/HFUI.js

```diff
@@ -18,26 +17,52 @@ import NotificationsSidebar from '../NotificationsSidebar'
 
 import Routes from '../../constants/routes'
 import { isElectronApp } from '../../redux/config'
+import MIN_SAFE_WIDTH from '../../constants/MIN_SAFE_WIDTH'
+import BestExperienceMessageModal from '../BestExperienceMessageModal'
 
 import './style.css'
 
+const HF_UI_WEB_SHOW_BEST_EXPERIENCE_MODAL = 'HF_UI_WEB_SHOW_BEST_EXPERIENCE_MODAL'
+
 const HFUI = ({
   authToken, getSettings, notificationsVisible, getFavoritePairs, currentMode, GAPageview,
   currentPage, onUnload, subscribeAllTickers, shouldShowAOPauseModalState, settingsShowAlgoPauseInfo,
 }) => {
-  function unloadHandler() {
+  const [isBestExperienceMessageModalOpen, setBestExperienceModalState] = useState(false)
+
+  const unloadHandler = () => {
     if (authToken !== null) {
       onUnload(authToken, currentMode)
     }
   }
 
-  function onElectronAppClose() {
+  const onElectronAppClose = () => {
     if (!authToken || !settingsShowAlgoPauseInfo) {
       closeElectronApp()
     } else {
       shouldShowAOPauseModalState()
     }
   }
+  const closeBestExperienceModal = () => setBestExperienceModalState(false)
+
+  const onSubmitBestExperienceModal = () => {
+    localStorage.setItem(HF_UI_WEB_SHOW_BEST_EXPERIENCE_MODAL, JSON.stringify(false))
+    closeBestExperienceModal()
+  }
+
+  useEffect(() => {
+    const currentWidth = document.documentElement.clientWidth
+    let needToShowModal = localStorage.getItem(HF_UI_WEB_SHOW_BEST_EXPERIENCE_MODAL)
+    if (needToShowModal === null || needToShowModal === undefined) {
+      needToShowModal = true
+    } else {
+      needToShowModal = JSON.parse(needToShowModal)
+    }
+    if (isElectronApp || !needToShowModal || currentWidth >= MIN_SAFE_WIDTH) {
```

Comment:
not sure if it's needed nowadays, but maybe we should put `if (isElectronApp || currentWIdth >= MIN_SAFE_WIDTH)` on top of `useEffect` (as a guard clause), so we would not have to to unnecessary actions related to `needToShowModal` in case if the `if conditions` above will return `true`